### PR TITLE
Update helm install

### PR DESCRIPTION
**Pull request by lizhuqi**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Can't use Helm to install driver for Windows because Windows does not support privileged containers. Change to use kubectl to install drivers. 

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
we should not use `https://raw.githubusercontent.com/kubernetes-csi/csi-driver-smb/master/deploy/install-driver.sh` for e2e test since it would install `latest` tag image which is not built by current code

**Comment by andyzhangx** [comment]
why not run the e2e test on a Linux machine?

**Comment by lizhuqi** [comment]
I am running the e2e test on a Linux machine. I can install the helm but when I try to use helm to install the CSI SMB driver, I got **Permission denied** error as below:
helm install csi-driver-smb charts/latest/csi-driver-smb --namespace kube-system --wait --timeout=15m -v=5 --debug \
        --set image.smb.repository=andyzhangx/smb-csi --set image.smb.tag=v0.7.0 \
        --set windows.enabled=true \
        --set linux.enabled=false \
        --set controller.replicas=1 \
        --set controller.logLevel=6 \
        --set node.logLevel=6
**make: helm: Permission denied**
make: *** [Makefile:97: e2e-bootstrap] Error 127

**Comment by andyzhangx** [comment]
> I am running the e2e test on a Linux machine. I can install the helm but when I try to use helm to install the CSI SMB driver, I got **Permission denied** error as below:
> helm install csi-driver-smb charts/latest/csi-driver-smb --namespace kube-system --wait --timeout=15m -v=5 --debug
> --set image.smb.repository=andyzhangx/smb-csi --set image.smb.tag=v0.7.0
> --set windows.enabled=true
> --set linux.enabled=false
> --set controller.replicas=1
> --set controller.logLevel=6
> --set node.logLevel=6
> **make: helm: Permission denied**
> make: *** [Makefile:97: e2e-bootstrap] Error 127

may add sudo? It's like config issue.

File: Makefile [hunk 1]
```diff
@@ -18,17 +18,20 @@ REGISTRY ?= andyzhangx
 REGISTRY_NAME = $(shell echo $(REGISTRY) | sed "s/.azurecr.io//g")
 IMAGE_NAME ?= smb-csi
 IMAGE_VERSION ?= v0.7.0
+WIN_IMAGE_VERSION ?= v0.6.0-windows-1809-amd64
```
Comment (andyzhangx) [inline Makefile:21]: should be `v0.7.0-windows-1809-amd64` which is `${IMAGE_VERSION}-windows-1809-amd64` ?

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

File: Makefile [hunk 2]
```diff
@@ -113,7 +119,7 @@ install-helm:
 
 .PHONY: e2e-teardown
 e2e-teardown:
-	helm delete csi-driver-smb --namespace kube-system
+	sudo /usr/local/bin/helm delete csi-driver-smb --namespace kube-system --kubeconfig ~/.kube/config
```
Comment (andyzhangx) [inline Makefile:122]: that's env config issue: `sudo /usr/local/bin/helm`, better not hardcode in Makefile

File: Makefile [hunk 3]
```diff
@@ -93,17 +96,20 @@ e2e-test:
 
 .PHONY: e2e-bootstrap
 e2e-bootstrap: install-helm
-	docker pull $(IMAGE_TAG) || make container-all push-manifest
 ifdef TEST_WINDOWS
-	helm install csi-driver-smb charts/latest/csi-driver-smb --namespace kube-system --wait --timeout=15m -v=5 --debug \
-		${E2E_HELM_OPTIONS} \
+	docker pull $(WIN_IMAGE_TAG) || make container-all push-manifest	
+	sudo /usr/local/bin/helm upgrade csi-driver-smb charts/latest/csi-driver-smb --namespace kube-system --wait --timeout=15m -v=5 --debug --install \
```
Comment (andyzhangx) [inline Makefile:101]: `helm upgrade`

File: Makefile [hunk 4]
```diff
@@ -93,17 +96,20 @@ e2e-test:
...
+	docker pull $(WIN_IMAGE_TAG) || make container-all push-manifest	
+	sudo /usr/local/bin/helm upgrade csi-driver-smb charts/latest/csi-driver-smb --namespace kube-system --wait --timeout=15m -v=5 --debug --install \
+		${E2E_WIN_HELM_OPTIONS} \
 		--set windows.enabled=true \
 		--set linux.enabled=false \
 		--set controller.replicas=1 \
 		--set controller.logLevel=6 \
-		--set node.logLevel=6
+		--set node.logLevel=6 \
+		--kubeconfig ~/.kube/config
```
Comment (andyzhangx) [inline Makefile:108]: I think `~/.kube/config` is the default `--kubeconfig` setting, not necessary to specify here

Comment (lizhuqi) [reply Makefile:108]: yeah, I thought so. However if I don't specify here, helm fails to connect to. the cluster.

No idea why... Can we keep it here?

File: Makefile (hunk 1 above)
Comment (lizhuqi) [reply Makefile]: Done. Thanks!

File: Makefile (hunk 2 above)
Comment (lizhuqi) [reply Makefile]: Ok. Removed those in the Makefile

**Comment by lizhuqi** [comment]
\retest

File: Makefile (hunk 4 above)
Comment (lizhuqi) [reply Makefile]: yeah, also removed --kubeconfig, it seems that it causes prow tests to fail

File: Makefile [hunk 5]
```diff
@@ -18,17 +18,20 @@ REGISTRY ?= andyzhangx
 REGISTRY_NAME = $(shell echo $(REGISTRY) | sed "s/.azurecr.io//g")
 IMAGE_NAME ?= smb-csi
 IMAGE_VERSION ?= v0.7.0
+WIN_IMAGE_VERSION ?= ${IMAGE_VERSION}-windows-1809-amd64
```
Comment (andyzhangx) [inline Makefile:21]: `v0.7.0` tag does not work on GKE Windows node? I think you could use one version tag for both Linux and Windows.

**Review by andyzhangx — COMMENTED** [review]
`image.smb.tag=e2e-2e4d133997830fbbc81f1283102c805706a0a9b9-windows-1809-amd64` does not work on Linux node, would you use same tag for both Linux and Windows when building image?

```
helm upgrade csi-driver-smb charts/latest/csi-driver-smb --namespace kube-system --wait --timeout=15m -v=5 --debug --install \
	--set image.smb.repository=k8sprow.azurecr.io/smb-csi --set image.smb.tag=e2e-2e4d133997830fbbc81f1283102c805706a0a9b9-windows-1809-amd64 \
	--set windows.enabled=true \
```

**Review by andyzhangx — COMMENTED** [review]
that's quite clear now, could you squash all commits? thanks.

**Comment by lizhuqi** [comment]
> that's quite clear now, could you squash all commits? thanks.

Finally squashed all the commits. Please take another look. Thanks!

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

File: Makefile [hunk 6]
```diff
@@ -95,15 +95,15 @@ e2e-test:
 e2e-bootstrap: install-helm
 	docker pull $(IMAGE_TAG) || make container-all push-manifest
 ifdef TEST_WINDOWS
-	helm install csi-driver-smb charts/latest/csi-driver-smb --namespace kube-system --wait --timeout=15m -v=5 --debug \
+	helm upgrade csi-driver-smb charts/latest/csi-driver-smb --namespace kube-system --wait --timeout=15m -v=5 --debug --install \
```
Comment (andyzhangx) [inline Makefile:98]: I found a bug for `helm upgrade`, agent node will always use `v0.7.0` in azure pipeline which breaks the pipeline finally, I would revert to use original `helm instal`
```
print out csi-smb-node logs ...
======================================================================================
[pod/csi-smb-node-sqjc2/smb] I0526 01:09:40.294321       1 main.go:85] set up prometheus server on [::]:29645
[pod/csi-smb-node-sqjc2/smb] I0526 01:09:40.294699       1 smb.go:62] 
[pod/csi-smb-node-sqjc2/smb] DRIVER INFORMATION:
[pod/csi-smb-node-sqjc2/smb] -------------------
[pod/csi-smb-node-sqjc2/smb] Build Date: "2021-05-22T01:05:27Z"
[pod/csi-smb-node-sqjc2/smb] Compiler: gc
[pod/csi-smb-node-sqjc2/smb] Driver Name: smb.csi.k8s.io
[pod/csi-smb-node-sqjc2/smb] Driver Version: v0.7.0
[pod/csi-smb-node-sqjc2/smb] Git Commit: a03a482b5e925a77edcdfbab2ae330be8f5866ff
[pod/csi-smb-node-sqjc2/smb] Go Version: go1.16
[pod/csi-smb-node-sqjc2/smb] Platform: linux/amd64
```

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_csi-driver-smb/265/pull-csi-driver-smb-e2e-windows/1397354833710682112/build-log.txt

Comment (andyzhangx) [reply Makefile:98]: nvm, it's not related, still try to figure out why agent node is always pulling image `v0.7.0`

**Pull request merged**